STM32/LightController/Core/Src/LightController.c

```diff
@@ -76,124 +114,150 @@ bool isInputValid(const char *input, int *channel, unsigned int *rgb)
 	return true;
 }
 
-int handleInput(unsigned int rgb, int *channel, uint8_t *red, uint8_t *green, uint8_t *blue)
+static int handleInput(unsigned int rgb, uint8_t *red, uint8_t *green, uint8_t *blue)
 {
 	*red = (rgb >> 16) & 0xFF;
 	*green = (rgb >> 8) & 0xFF;
 	*blue = rgb & 0xFF;
 	return (rgb == 0xFFFFFF) ? 1 : 0; // If 0xFFFFFF turn on the white in other ports
 }
 
-TIM_HandleTypeDef * htim2_ = NULL;
-TIM_HandleTypeDef * htim3_ = NULL;
-TIM_HandleTypeDef * htim4_ = NULL;
-static void updateLED(int channel, unsigned int red, unsigned int green, unsigned int blue)
+
+static void updateLEDCtrl(int channel, unsigned int red, unsigned int green, unsigned int blue, int whiteOn)
 {
-	TIM_HandleTypeDef * portHandle = NULL;
-	switch (channel)
+	if (whiteOn)
 	{
-		case 1:
-		portHandle = htim2_;
-		break;
-
-		case 2:
-		portHandle = htim3_;
-		break;
-
-		case 3:
-		portHandle = htim4_;
-		break;
+		rgbwControl[channel*NO_COLORS] 	   = 0;
+		rgbwControl[channel*NO_COLORS + 1] = 0;
+		rgbwControl[channel*NO_COLORS + 2] = 0;
+		rgbwControl[channel*NO_COLORS + 3] = 256; 
```

Comment:
We chose to not update the input format with the extra hex value for the white color to not need to different formats to be supported by LoopCode. 

It does limit the functionality of the white color slightly. It could maybe be parsed such that equal numbers on the RGB colors result in that specific on time for the white color such that the intensity can be chosen freely.

Reply:
The 256 is a bug - I've updated it to 255.